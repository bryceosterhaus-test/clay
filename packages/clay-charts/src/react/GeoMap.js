import React from 'react';
import {Geomap} from '../util/geo-map';

export default class GeomapReact extends React.Component {
	constructor(props) {
		super(props);

		this._containerRef = React.createRef();
	}
	componentDidMount() {
		this._geoMapInstance = new Geomap({
			...this.props,
			element: this._containerRef.current,
		});

		this._geoMapInstance.attached();
	}

	/**
	 * @inheritDoc
	 */
	componentWillUnmount() {
		this._geoMapInstance.disposed();
	}

	render() {
		const {data, element, color, pollingInterval, ...otherProps} = this.props;

		const {height = '100%', width = '100%'} = this._geoMapInstance
			? this._geoMapInstance.getSize()
			: {};

		return (
			<div
				style={{height, width}}
				{...otherProps}
				ref={this._containerRef}
			/>
		);
	}
}
