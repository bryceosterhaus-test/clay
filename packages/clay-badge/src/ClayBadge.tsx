/**
 * Â© 2019 Liferay, Inc. <https://liferay.com>
 *
 * SPDX-License-Identifier: BSD-3-Clause
 */
import * as React from 'react';
import classNames from 'classnames';

interface Props extends React.HTMLAttributes<HTMLSpanElement> {
	displayType?: string;
	label?: string | number;
}

const ClayBadge: React.FunctionComponent<Props> = ({
	className,
	displayType,
	label,
	...otherProps
}) => (
	<span
		{...otherProps}
		className={classNames('badge', `badge-${displayType}`, className)}
	>
		<span className="badge-item badge-item-expand">{label}</span>
	</span>
);

ClayBadge.defaultProps = {
	displayType: 'primary',
};

export default ClayBadge;