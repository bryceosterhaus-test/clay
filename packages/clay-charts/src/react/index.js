import React from 'react';
import BillboardChart from 'react-billboardjs';
import GeoMap from './react/GeoMap';

import {
	DEFAULT_COLORS,
	DEFAULT_GRID_OBJECT,
	DEFAULT_LINE_CLASSES,
	DEFAULT_POINT_PATTERNS,
} from '../ChartBase';

import '../scss/main.scss';

export default class extends React.Component {
	render() {
		const {color, data, grid, line, point, ...otherProps} = this.props;

		let ChartComponent;
		let props = {data, ...otherProps};

		switch (data.type) {
			case 'geo-map':
				ChartComponent = GeoMap;
				break;
			default:
				ChartComponent = BillboardChart;
		}

		return (
			<ChartComponent
				{...otherProps}
				data={data}
				grid={Object.assign(DEFAULT_GRID_OBJECT, grid)}
				color={Object.assign({pattern: DEFAULT_COLORS}, color)}
				line={Object.assign({classes: DEFAULT_LINE_CLASSES}, line)}
				point={Object.assign(
					{
						pattern: DEFAULT_POINT_PATTERNS,
					},
					point
				)}
			/>
		);
	}
}
