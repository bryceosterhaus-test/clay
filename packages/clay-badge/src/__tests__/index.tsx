import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import ClayBadge from '..';

describe('ClayBadge', function() {
	it('should render', () => {
		const testRenderer = TestRenderer.create(
			<ClayBadge type="primary" label="4" />
		);

		expect(testRenderer.toJSON()).toMatchSnapshot();
	});

	it('should render with custom className', () => {
		const testRenderer = TestRenderer.create(
			<ClayBadge className="foo bar" type="primary" label="4" />
		);

		expect(testRenderer.toJSON()).toMatchSnapshot();
	});
});
