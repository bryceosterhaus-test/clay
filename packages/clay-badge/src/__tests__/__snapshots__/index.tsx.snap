// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClayBadge should render 1`] = `
<span
  className="badge badge-primary"
>
  <span
    className="badge-item badge-item-expand"
  >
    4
  </span>
</span>
`;

exports[`ClayBadge should render with custom className 1`] = `
<span
  className="badge badge-primary foo bar"
>
  <span
    className="badge-item badge-item-expand"
  >
    4
  </span>
</span>
`;
